### PR TITLE
OSD-9708 Separate OA config cr fields from OAO cr fields

### PR DESCRIPTION
**Pull request by anjoshi24**
Signed-off-by: Aniket Joshi <anjoshi@redhat.com>

### What type of PR is this?
Refactor.

### Which Jira/Github issue(s) this PR fixes?
[OSD-9708](https://issues.redhat.com/browse/OSD-9708)

_Fixes #_

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/apis/ocmagent/v1alpha1/ocmagent_types.go [hunk 1]
```diff
@@ -4,17 +4,22 @@ import (
...
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	// OcmBaseUrl defines the OCM api endpoint for OCM agent to access
-	OcmBaseUrl string `json:"ocmBaseUrl"`
-
-	// Services defines the supported OCM services, eg, service_log, cluster_management
-	Services []string `json:"services"`
+	// AgentConfig refers to OCM agent config fields separated
+	Config AgentConfig `json:"agentConfig"`
```
Comment (mrbarge) [inline pkg/apis/ocmagent/v1alpha1/ocmagent_types.go:22]: Maybe call this `AgentConfig` to keep it consistent with naming of the json field?

**Comment by mrbarge** [comment]
Can you also update the example test CR (https://github.com/anjoshi24/ocm-agent-operator/blob/OSD-9708/test/deploy/60_ocmagent.OcmAgent.yaml) to reflect this change?

File: pkg/apis/ocmagent/v1alpha1/ocmagent_types.go (hunk 1 above)
Comment (anjoshi24) [reply pkg/apis/ocmagent/v1alpha1/ocmagent_types.go]: Thanks for the review, made the change as per the suggestion.

**Comment by anjoshi24** [comment]
Updated the test CR to reflect the changes as suggested.

File: test/deploy/60_ocmagent.OcmAgent.yaml [hunk 2]
```diff
@@ -4,10 +4,11 @@ metadata:
   name: ocmagent
   namespace: test-ocm-agent-operator
 spec:
-  ocmBaseUrl: "https://api.stage.openshift.com"
-  replicas: 1
-  services:
+  agentConfig:
+    ocmBaseUrl: "https://api.stage.openshift.com"
+    services:
   - service_log
```
Comment (mrbarge) [inline test/deploy/60_ocmagent.OcmAgent.yaml:10]: Missing an indent, this results in invalid yaml

Comment (anjoshi24) [reply test/deploy/60_ocmagent.OcmAgent.yaml]: Thanks, fixed it!

**Pull request merged**